src/core/core.controller.js

```diff
@@ -933,7 +973,26 @@ module.exports = function(Chart) {
 			me.lastActive = me.active;
 
 			return changed;
-		}
+		},
+		/**
+		 * Format the data list if the data is missing some points in the datasets.
+		 * @private
+		 * @param {array} dataArray array to format
+		 * @return {array} the formated array
+		 */
+		_formatDataset: function(dataArray) {
+			var labels = this.chart.data.labels;
+			var tmp = dataArray.slice(0);
+			var result = dataArray;
+			var match = dataInLabel(tmp, labels);
+			if (match && dataArray.length < labels.length) {
+				for (var i = 0; i < labels.length; i++) {
+					var label = labels[i];
+					result[i] = {x: label, y: getY(label, tmp)};
```

Comment:
I think you need to check if it's a horizontal or vertical chart here and can't just assume that the label will be on the x-axis

And `getY` should probably be called `getValue` and also do the same check